xen.spec.in

```diff
@@ -748,61 +694,89 @@ fi
 if [ $1 == 1 -a -f /sbin/grub2-mkconfig ]; then
   if [ -f /boot/grub2/grub.cfg ]; then
     /sbin/grub2-mkconfig -o /boot/grub2/grub.cfg
+    sed -i -e '/insmod module2/d' /boot/grub2/grub.cfg
   fi
   if [ -f /boot/efi/EFI/qubes/grub.cfg ]; then
     /sbin/grub2-mkconfig -o /boot/efi/EFI/qubes/grub.cfg
+    sed -i -e '/insmod module2/d' /boot/efi/EFI/qubes/grub.cfg
+  fi
+fi
+if [ -f /sbin/grub2-mkconfig ]; then
+  if [ -f /boot/grub2/grub.cfg ]; then
+    if [ -d /usr/lib/grub/i386-pc ]; then
+      if [ ! -d /boot/grub2/i386-pc ]; then
+        mkdir /boot/grub2/i386-pc
+      fi
+      if [ -f /usr/lib/grub/i386-pc/relocator.mod -a ! -f /boot/grub2/i386-pc/relocator.mod ]; then
+        cp -p /usr/lib/grub/i386-pc/relocator.mod /boot/grub2/i386-pc/relocator.mod
+      fi
+      if [ -f /usr/lib/grub/i386-pc/multiboot2.mod -a ! -f /boot/grub2/i386-pc/multiboot2.mod ]; then
+        cp -p /usr/lib/grub/i386-pc/multiboot2.mod /boot/grub2/i386-pc/multiboot2.mod
+      fi
+    fi
+  fi
+  if [ -f /boot/efi/EFI/qubes/grub.cfg ]; then
+    if [ -d /usr/lib/grub/x86_64-efi ]; then
+      if [ ! -d /boot/efi/EFI/qubes/x86_64-efi ]; then
+        mkdir /boot/efi/EFI/qubes/x86_64-efi
+      fi
+      if [ -f /usr/lib/grub/x86_64-efi/relocator.mod -a ! -f /boot/efi/EFI/qubes/x86_64-efi/relocator.mod ]; then
+        cp -p /usr/lib/grub/x86_64-efi/relocator.mod /boot/efi/EFI/qubes/x86_64-efi/relocator.mod
+      fi
+      if [ -f /usr/lib/grub/x86_64-efi/multiboot2.mod -a ! -f /boot/efi/EFI/qubes/x86_64-efi/multiboot2.mod ]; then
+        cp -p /usr/lib/grub/x86_64-efi/multiboot2.mod /boot/efi/EFI/qubes/x86_64-efi/multiboot2.mod
+      fi
+    fi
   fi
 fi
 
 %postun hypervisor
 if [ -f /sbin/grub2-mkconfig ]; then
   if [ -f /boot/grub2/grub.cfg ]; then
     /sbin/grub2-mkconfig -o /boot/grub2/grub.cfg
+    sed -i -e '/insmod module2/d' /boot/grub2/grub.cfg
   fi
   if [ -f /boot/efi/EFI/qubes/grub.cfg ]; then
     /sbin/grub2-mkconfig -o /boot/efi/EFI/qubes/grub.cfg
+    sed -i -e '/insmod module2/d' /boot/efi/EFI/qubes/grub.cfg
   fi
 fi
 %endif
 
 %if %build_ocaml
 %post ocaml
+%if %with_systemd_presets
 %systemd_post oxenstored.service
+%else
+if [ $1 == 1 ]; then
+  /bin/systemctl enable oxenstored.service
+fi
+%endif
 
 %preun ocaml
+%if %with_systemd_presets
 %systemd_preun oxenstored.service
+%else
+if [ $1 == 0 ]; then
+  /bin/systemctl disable oxenstored.service
+fi
+%endif
 %endif
-
-%posttrans libs
-# reload libxl library
-systemctl try-restart libvirtd.service >/dev/null 2>&1 || :
```

Comment:
This part was needed somewhere, otherwise libvirt will use old libraries after update. But it can be moved as a trigger into libvirt.spec (requires testing).

Reply:
Ok I will add it to libvirt.spec.in